### ISSUE TITLE
module/expand-path removes dots from Unix hidden directory names

### ISSUE THREAD

**Issue by stutonk** [janet-lang/janet#143]
Using a build from the latest master HEAD on Linux:

Setting `JANET_PATH` to a directory with a Unix-style hidden name (e.g. `.janet/`) causes module imports to fail due to removal of dots from hidden directory names as in the following transcript from the REPL
```
$ JANET_PATH=".janet/.janet/" janet
Janet 1.1.0-dev-21a355c  Copyright (C) 2017-2019 Calvin Rose
janet:0:> (dyn :syspath)
".janet/.janet/"
janet:15:>  (module/expand-path "somelib" ":sys:/:all:.janet")
@"janet/janet/somelib.janet"

```

The problem seems to be in `janet_core_expand_path` located at src/core/corelib.c:118, presumably in the section starting on line 180, which I don't understand well enough at a cursory glance to see what the exact problem is.

**Comment by bakpakin** [comment]
Yes, seems to be a bug. This function has the dual role of substituting a few templated values and normalizing a path for import. The logic is supposed to convert a/./b to a/b, and it seems it erroneously chopped of a dot that was not a path segment. I will look more at this tonight.

**Comment by bakpakin** [comment]
This should be fixed as of e0fe847, @stutonk, please close if it you can verify that it works.

**Comment by stutonk** [comment]
Confirmed that it now works as expected. Thanks for the quick fix.
